Fix plan output relative

james03160927 opened the pull request:
<!-- Prepend '[WIP]' to the title if this PR is still a work-in-progress. Remove it when it is ready for review! -->

## Description

Fixes https://github.com/gruntwork-io/terragrunt/issues/4430. 

This pull request introduces enhancements to the handling of module paths, adds new test fixtures for validating Terragrunt plans, and implements a new integration test for verifying `terragrunt plan --all` functionality. Below is a summary of the most important changes:

**Enhancements to module path handling:** Updated `getPlanFilePath` in `configstack/module.go` to ensure `opts.WorkingDir` is converted to an absolute path before calling `filepath.Rel`, improving compatibility with absolute module paths.

<!-- Description of the changes introduced by this PR. -->

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [ ] I authored this code entirely myself
- [ ] I am submitting code based on open source software (e.g. MIT, MPL-2.0, Apache)]
- [ ] I am adding or upgrading a dependency or adapted code and confirm it has a compatible open source license
- [ ] Update the docs.
- [ ] Run the relevant tests successfully, including pre-commit checks.
- [ ] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

<!-- One-line description of the PR that can be included in the final release notes. -->
Added / Removed / Updated [X].

### Migration Guide

<!-- Important: If you made any backward incompatible changes, then you must write a migration guide! -->

denis256 commented:
Looks like are failing tests related to `TestTfPath`

yhakbar commented:
> `TestTfPath`

Might be my fault. Let me update the PR and see if that fixes it.

james03160927 commented:
I don't see `TestTfPath` test failing when I search for "FAIL:" from the failing checks below. @yhakbar

james03160927 commented:
HI @denis256 @yhakbar, should be good for another review.

denis256 commented:
Looks like it is failing on lint

```
test/integration_test.go:4263:1: Function TestTerragruntPlanAllOutput missing the call to method parallel (paralleltest)
func TestTerragruntPlanAllOutput(t *testing.T) {
^
internal/runner/common/unit.go:98:2: assignments should only be cuddled with other assignments (wsl)
	path, err := filepath.Rel(wdir, unit.Path)
	^
internal/runner/common/unit.go:99:2: only one cuddle assignment allowed before if statement (wsl)
	if err != nil {
	^
3 issues:
* paralleltest: 1
* wsl: 2
level=info msg="Memory: 626 samples, avg is 1971.2MB, max is 3164.3MB"
```

@james03160927 ^

james03160927 commented:
> Looks like it is failing on lint
> 
> ```
> test/integration_test.go:4263:1: Function TestTerragruntPlanAllOutput missing the call to method parallel (paralleltest)
> func TestTerragruntPlanAllOutput(t *testing.T) {
> ^
> internal/runner/common/unit.go:98:2: assignments should only be cuddled with other assignments (wsl)
> 	path, err := filepath.Rel(wdir, unit.Path)
> 	^
> internal/runner/common/unit.go:99:2: only one cuddle assignment allowed before if statement (wsl)
> 	if err != nil {
> 	^
> 3 issues:
> * paralleltest: 1
> * wsl: 2
> level=info msg="Memory: 626 samples, avg is 1971.2MB, max is 3164.3MB"
> ```
> 
> @james03160927 ^

Hi @denis256, one of the previous lint comment suggested to use `t.Chdir()` and it seems like we cannot use `t.Parallel()` if I follow the previous lint suggestion. Do you prefer me to use t.Parallel() and ignore the comment about using t.Chdir()`

yhakbar commented:
> > Looks like it is failing on lint
> > ```
> > test/integration_test.go:4263:1: Function TestTerragruntPlanAllOutput missing the call to method parallel (paralleltest)
> > func TestTerragruntPlanAllOutput(t *testing.T) {
> > ^
> > internal/runner/common/unit.go:98:2: assignments should only be cuddled with other assignments (wsl)
> > 	path, err := filepath.Rel(wdir, unit.Path)
> > 	^
> > internal/runner/common/unit.go:99:2: only one cuddle assignment allowed before if statement (wsl)
> > 	if err != nil {
> > 	^
> > 3 issues:
> > * paralleltest: 1
> > * wsl: 2
> > level=info msg="Memory: 626 samples, avg is 1971.2MB, max is 3164.3MB"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > @james03160927 ^
> 
> Hi @denis256, one of the previous lint comment suggested to use `t.Chdir()` and it seems like we cannot use `t.Parallel()` if I follow the previous lint suggestion. Do you prefer me to use t.Parallel() and ignore the comment about using t.Chdir()`
> 
> Original change here: [dc489ee](https://github.com/gruntwork-io/terragrunt/commit/dc489ee459dcd3a19544d9fce992878ef23f71f3)

Why use chdir at all instead of using `working-dir`?

pseudomorph commented:
@yhakbar - Might it make more sense to add a setter to the WorkingDir flag? The default working dir is an absolute path when not specified.

Seems like there are a few different issues related to the abs/rel pathing for working dir.

pseudomorph commented:
I've figured a top-level fix which allows the tests here to pass and addresses issues I was seeing with using `--working-dir` with the `find` command.

I'll cut an issue for the `find` matter. Should I post a patch here, or cut a different PR?

james03160927 commented:
> I've figured a top-level fix which allows the tests here to pass and addresses issues I was seeing with using `--working-dir` with the `find` command.
> 
> I'll cut an issue for the `find` matter. Should I post a patch here, or cut a different PR?

OH interesting. Feel free to contribute to this PR and we can work on things together. Thanks a lot @pseudomorph
